src/main/java/uk/gov/hmcts/reform/em/stitching/batch/DocumentTaskItemProcessor.java

```diff
@@ -69,25 +81,25 @@ public DocumentTask process(DocumentTask documentTask) {
 
             if (StringUtils.isNotBlank(documentTask.getCaseTypeId())
                 && StringUtils.isNotBlank(documentTask.getJurisdictionId())) {
-                Map<BundleDocument, File> bundleFiles = cdamService
+                bundleFiles = cdamService
                     .downloadFiles(documentTask)
                     .map(unchecked(documentConverter::convert))
                     .map(file -> pdfWatermark.processDocumentWatermark(documentImage, file, documentTask.getBundle().getDocumentImage()))
                     .collect(Collectors.toMap(Pair::getFirst, Pair::getSecond));
                 log.info("Documents downloaded through CDAM for DocumentTask Id : #{} ", documentTask.getId());
-                final File outputFile = pdfMerger.merge(documentTask.getBundle(), bundleFiles, coverPageFile);
+                outputFile = pdfMerger.merge(documentTask.getBundle(), bundleFiles, coverPageFile);
```

Comment:
I think we can try "DocumentMergeMode"  to `OPTIMIZE_RESOURCES_MODE`
`OPTIMIZE_RESOURCES_MODE Optimizes resource handling such as closing documents early. Not all document elements are merged compared to the PDFBOX_LEGACY_MODE. Currently supported are:
Page content and resources`

Reply:
This fix was first iteration to resolve the memory issue, which it has drastically brought down the memory consumption. We can try your above solution may be second iteration.